kubernetes/fineractmysql-deployment.yml

```diff
@@ -0,0 +1,110 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements. See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership. The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: fineractmysql-pv-volume
+  labels:
+    type: local
+spec:
+  storageClassName: manual
+  capacity:
+    storage: 10Gi
+  accessModes:
+    - ReadWriteMany
+  hostPath:
+    path: "/mnt/data"
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: fineractmysql-pv-claim
+spec:
+  storageClassName: manual
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 5Gi
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: fineract-server
+  name: fineractmysql
+spec:
+  ports:
+    - protocol: TCP
+      port: 3306
+      targetPort: 3306
+  selector:
+    app: fineract-server
+    tier: fineractmysql
+  clusterIP: None
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: fineract-server
+  name: fineractmysql
+spec:
+  selector:
+    matchLabels:
+      app: fineract-server
+      tier: fineractmysql
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app: fineract-server
+        tier: fineractmysql
+    spec:
+      containers:
+        - image: mysql:5.7
+          name: mysql
+          env:
+            - name: MYSQL_ROOT_PASSWORD
+              value: skdcnwauicn2ucnaecasdsajdnizucawencascdca
+          livenessProbe:
+            exec:
+              command:
+                - mysqladmin
+                - ping
+                - -h
+                - localhost
+            failureThreshold: 10
+            timeoutSeconds: 10
+          ports:
+            - containerPort: 3306
+              name: fineractmysql
+          volumeMounts:
+            - name: fineractmysql-persistent-storage
+              mountPath: /var/lib/mysql/
+            - name: fineractmysql-host-path
+              mountPath: /docker-entrypoint-initdb.d/
+      volumes:
+        - name: fineractmysql-persistent-storage
+          persistentVolumeClaim:
+            claimName: fineractmysql-pv-claim
+        - name: fineractmysql-host-path
+          hostPath:
+            path: /fineract-db
```

Comment:
I see what you are doing, but it seems like a little bit of a shame to me that JUST for 2 lines of CREATE DB we have to do the whole mount thing... How about using a ConfigMap PV for this instead? This could also be done in a follow up PR, of course.

Reply:
I had some trouble using configMap, so with all the failures from before, I tried to focus on a solution that works first. And then lateron modifications can be done.

Reply:
Sure, SGTM. Do you want to create a JIRA ticket for future follow up on this? Then Resolve.

Reply:
Sure